tests/system/src/test/java/test/robot/javafx/embed/swing/JFXPanelScaledTest.java

```diff
@@ -0,0 +1,151 @@
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.embed.swing;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static test.util.Util.TIMEOUT;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.image.BufferedImage;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.Timer;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import javafx.embed.swing.JFXPanelShim;
+import javafx.scene.Scene;
+import javafx.scene.layout.Region;
+import junit.framework.AssertionFailedError;
+
+public class JFXPanelScaledTest {
+    static CountDownLatch launchLatch;
+
+    private static MyApp myApp;
+
+    private static Timer t;
+    static int cnt;
+
+    @BeforeClass
+    public static void setupOnce() throws Exception {
+        System.setProperty("sun.java2d.uiScale.enabled", "true");
+        System.setProperty("sun.java2d.uiScale", "125%");
+        System.setProperty("glass.win.uiScale", "1.25");
+        System.setProperty("glass.gtk.uiScale", "1.25");
+        launchLatch = new CountDownLatch(1);
+
+        // Start the Application
+        SwingUtilities.invokeLater(() -> {
+            myApp = new MyApp();
+        });
+        if (!launchLatch.await(5 * TIMEOUT, TimeUnit.MILLISECONDS)) {
+            throw new AssertionFailedError("Timeout waiting for Application to launch (" + (5 * TIMEOUT) + " seconds)");
+        }
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        if (myApp != null) {
+            SwingUtilities.invokeLater(myApp::dispose);
+        }
+    }
+
+    @Test
+    public void testScale() throws Exception {
+        // Get the Swing-side BackBuffer
+        BufferedImage pixelsIm = JFXPanelShim.getPixelsIm(myApp.jfxPanel);
+        assertEquals(127, pixelsIm.getWidth());
+        assertEquals(127, pixelsIm.getHeight());
```

Comment:
Where does the `127` come from? If this is derived from the size of the JFXPanel * scale, it isn't likely to work on platforms other than Windows (it certainly won't work on Mac, and I suspect not on Linux, but it needs to be tested). Also, unless there is a padding of 1 pixel (in user space coordinates), wouldn't it be 125?

Reply:
I'm using a base size of 101, so that there is a difference between Math.ceil and Math.round for width*1.25.
100 is only the initial size. The Timer changes the bounds first to 201 and then to 101. Without resizing the issue does not appear.
I believe somewhere in the Pulse-Thread? there is also a creation/resizing of the Buffer using Math.ceil. I'm trying to clarify that now using constants.

My main concern is that the conversion vom JFX to Swing (and SWT, as you mentioned) is consistent, so there are no wrong line offsets when reading the buffer outside of JFX.

Reply:
OK. You can update the PR with your findings.